lib/provider/downloader_provider.dart

```diff
@@ -356,6 +370,44 @@ class DownloadProvider extends ChangeNotifier {
 
     log('Downloading to: $savePath ($baseDirectory)');
 
+    if (Platform.isLinux) {
+      if (baseDirectory == BaseDirectory.root) {
+        if (!savePath.startsWith('/')) {
+          savePath = '/$savePath';
+        }
+      } else {
+        final appDataDir = await getApplicationSupportDirectory();
```

Comment:
I don’t think this directory is the right one, as:

> Path to a directory where the application may place application support files.  
> Use this for files you don’t want exposed to the user. Your app should not use this directory for user data files.  

I am not sure if we should implement this check. We should check for permissions, but this should be centralized. I will check how we can handle this across multiple platforms. 
Also, editing the permissions doesn’t seem like a good idea. If the user decided that this directory should not be owned by them or should have no write access, we should not overwrite it. Instead, we should clearly indicate that we don’t have permission or, at the very least, make the overwrite an optional action.